internal/restic/snapshot_policy.go

```diff
@@ -260,13 +260,16 @@ func ApplyPolicy(list Snapshots, p ExpirePolicy) (keep, remove Snapshots, reason
 
 		// Now update the other buckets and see if they have some counts left.
 		for i, b := range buckets {
-			if b.Count > 0 {
+			if b.Count <= -1 || b.Count > 0 {
```

Comment:
`b.Count == -1` should also be enough? (The current condition is just a complicated way of saying `b.Count != 0`.) And please add a comment, that -1 means infinite.